### ISSUE TITLE
Rename `entry_schema` and `key_schema` to `entry_type` and `key_type`

### ISSUE THREAD

**Issue by tliron** [oasis-open/tosca-community-contributions#125]
The word "schema" is confusing here, as we are referring to a data type.

Changing it to `_type` also aligns with my general preference that keynames include the name of entity type of what they refer to.

**Comment by lauwers** [comment]
I don't believe "type" is the correct type of entity here. Let's assume we have a list of lists. In that case the "entry_schema" is "list". But since "entry_schema" is required for lists, we also need to specify the entry schema for the list type specified in the original entity schema. This means that we may need full parameter definition syntax here, in which case we should consider renaming entry_schema to entry_definition, not entry_type.
